packages/edit-site/src/components/sidebar-edit-mode/navigation-menu-sidebar/navigation-inspector.js

```diff
@@ -17,38 +17,24 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import NavigationMenu from './navigation-menu';
-import { useHistory } from '../../routes';
 
 const NAVIGATION_MENUS_QUERY = [ { per_page: -1, status: 'publish' } ];
 
-function NavigationEffect() {
-	const history = useHistory();
-	const { postType, postId } = useSelect( ( select ) => {
+function SelectionEffect( { onSelect } ) {
+	const selectedBlock = useSelect( ( select ) => {
 		const { getSelectedBlock } = select( blockEditorStore );
-		const selectedBlock = getSelectedBlock();
-		if ( ! selectedBlock ) {
-			return {};
-		}
-		const { attributes } = selectedBlock;
-		if (
-			attributes.kind === 'post-type' &&
-			attributes.id &&
-			attributes.type
-		) {
-			return { postType: attributes.type, postId: attributes.id };
-		}
-		return {};
+		return getSelectedBlock();
 	} );
-	// When a navigation item liking to a post is selected, navigate on the site editor.
+	// When a navigation item is selected call the onSelect callback.
 	useEffect( () => {
-		if ( history && postType && postId ) {
-			history.push( { postType, postId } );
+		if ( selectedBlock ) {
+			onSelect( selectedBlock );
```

Comment:
Is there a way to call this in `onClick` events instead. Could the `onSelect` prop be passed down to `NavigationMenu` ...?

Reply:
Hi @youknowriad, I followed this suggestion and no onSelect is called on the click event instead of relying on an effect.